### PR TITLE
Run3-gex161 Some technical changes to avoid compilation warnings in a number of simulation packages

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Some technical changes to avoid compilation warnings in a number of simulation packages

#### PR validation:

Use the runTheMatrix test workflows

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
I shall try it

**Comment by civanch** [comment]
+1

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: SimG4CMS/HcalTestBeam/plugins/HcalTB04Analysis.cc [hunk 1]
```diff
@@ -987,13 +992,15 @@ void HcalTB04Analysis::fillEvent(PHcalTB04Info& product) {
...
                                   << " " << std::setw(1) << z << " " << std::setw(3) << ieta << " " << std::setw(3)
                                   << iphi << " T " << std::setw(6) << itr->t() << " E " << std::setw(6) << itr->e();
+    ++i;
 #endif
   }
   edm::LogVerbatim("HcalTBSim") << "HcalTB04Analysis:: Saves " << nhit << " hits from Crystals";
-  int hit = nhit;
+#ifdef EDM_ML_DEBUG
+  i = nhit;
+#endif
```
[the remark this answers is not in the document]
Comment (bsunanda) [reply SimG4CMS/HcalTestBeam/plugins/HcalTB04Analysis.cc]: Taken care of

File: SimG4CMS/HcalTestBeam/plugins/HcalTB04Analysis.cc [hunk 2]
```diff
@@ -987,13 +992,15 @@ void HcalTB04Analysis::fillEvent(PHcalTB04Info& product) {
                                   << group << std::dec << " " << std::setw(2) << det << " " << std::setw(2) << lay
                                   << " " << std::setw(1) << z << " " << std::setw(3) << ieta << " " << std::setw(3)
                                   << iphi << " T " << std::setw(6) << itr->t() << " E " << std::setw(6) << itr->e();
+    ++i;
```
[the remark this answers is not in the document]
Comment (bsunanda) [reply SimG4CMS/HcalTestBeam/plugins/HcalTB04Analysis.cc]: Taken care of

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by civanch** [comment]
+1

**Pull request merged**